test/cases/array.rb

```diff
@@ -1,56 +1,89 @@
 # frozen_string_literal: true
 
-# rubocop:disable Lint/Void
-
-[]
-
-[1, 2, 3]
-
-['a', 'b', 'c']
-
-['a', 'b c', 'd']
-
-[:a, :b, :c]
-
-[:a, :"b c", :d]
-
-%w[a b c]
-
-%i[a b c]
-
-%W[a#{a}a b#{b}a c#{c}c]
-
-%I[a#{a}a b#{b}b c#{c}c]
-
-# rubocop:disable Lint/UnneededSplatExpansion
-[1, 2, *[3, 4], 5, 6]
-# rubocop:enable Lint/UnneededSplatExpansion
-
-[
-  super_super_super_super_super_super_super_super_super_super_super_long,
-  super_super_super_super_super_super_super_super_super_super_super_long, [
-    super_super_super_super_super_super_super_super_super_super_super_long,
-    super_super_super_super_super_super_super_super_super_super_super_long
-  ]
-]
-
-a[1]
-
-a[super_super_super_super_super_super_super_super_super_super_super_super_sulong]
-
-a[1] = 2
-
-a[super_super_super_super_super_super_super_super_super_super_super_super_sulong] =
-  super_super_super_super_super_super_super_super_super_super_super_super_long
-
-a[1] = [
-  super_super_super_super_super_super_super_super_super_super_super_long,
-  super_super_super_super_super_super_super_super_super_super_super_long
-]
-
-a[1] = [
-  # abc
-  %w[abc]
-]
-
-# rubocop:enable Lint/Void
+# rubocop:disable Metrics/LineLength
+# rubocop:disable Metrics/MethodLength
+# rubocop:disable Metrics/AbcSize
+
+class ArrayTest < Minitest::Test
+  def test_array
+    assert_equal [], []
+    assert_equal [1, 2, 3], [1, 2, 3]
+    assert_equal ['a', 'b', 'c'], ['a', 'b', 'c']
+    assert_equal ['a', 'b c', 'd'], ['a', 'b c', 'd']
+    assert_equal [:a, :b, :c], [:a, :b, :c]
+    assert_equal [:a, :"b c", :d], [:a, :"b c", :d]
+  end
+
+  def test_array_with_general_delimited_syntax
+    a = 'foo'
+    b = 'bar'
+    c = 'baz'
+    assert_equal %w[a b c], %w[a b c]
```

Comment:
@kddeisz Will definitely have more context here - but I think we'll want something more static on the left hand side to verify against - that way this test isn't unintentionally influenced by prettier

For instance, if there was a bug in the formatter that incorrectly converted the`%w[ ... ]` notation into an empty array `[]`, after prettier runs on this file it would be update both of the arguments to be:

```diff
- assert_equal %w[a b c], %w[a b c]
+ assert_equal [], []
```

The above test would still pass, but we probably didn't want that!

If the left hand side was a static string to verify against though, it would've caught this. So as a small example the following test wouldn't break in the above way:

```
assert_equal 'a,b,c', %w[a b c].join(',')
```

As after prettier runs with the bug it would be converted to the following format, and the test would fail and catch the bug:

```diff
- assert_equal 'a,b,c', %w[a b c].join(',')
+ assert_equal 'a,b,c', [].join(',')
```

Check out the approach used within https://github.com/prettier/plugin-ruby/blob/29447d8a8514fe54fe28f571d9d9c7463a791736/test/cases/numbers.rb#L5 as a similar example

Let me know your thoughts 👍

Reply:
@AlanFoster, thanks for the explanation! that makes a lot more sense.